src/shared/ethtool-util.c

```diff
@@ -498,22 +501,37 @@ static int get_stringset(int ethtool_fd, struct ifreq *ifr, int stringset_id, st
         return 0;
 }
 
-static int find_feature_index(struct ethtool_gstrings *strings, const char *feature) {
-        unsigned i;
+static int set_features_bit(
+                const struct ethtool_gstrings *strings,
+                const char *feature,
+                bool flag,
+                struct ethtool_sfeatures *sfeatures) {
+        bool found = false;
 
-        for (i = 0; i < strings->len; i++) {
-                if (streq((char *) &strings->data[i * ETH_GSTRING_LEN], feature))
-                        return i;
-        }
+        assert(strings);
+        assert(feature);
+        assert(sfeatures);
+
+        for (size_t i = 0; i < strings->len; i++)
+                if (fnmatch(feature, (char *) &strings->data[i * ETH_GSTRING_LEN], 0) == 0) {
```

Comment:
hmm, do we really need globs here? maybe just do a prefix match if feature ends with "-"?

```c
if (streq((char *) &strings->data[i * ETH_GSTRING_LEN], feature) ||
        endswith(feature, "-") && startswith(strings->data[i * ETH_GSTRING_LEN], feature)) …
```

if you get what i mean...

I mean, globbing appears real overkill if all we care about is a prefix match